### ISSUE TITLE
Curl not printed when httpClient dontReuseHttpClientInstance() appended to config

### ISSUE THREAD

**Issue by francislainy** [dzieciou/curl-logger#61]
I have this sample class where I can have a curl printed.

    import com.github.dzieciou.testing.curl.CurlRestAssuredConfigFactory;
    import com.github.dzieciou.testing.curl.Options;
    import io.restassured.config.RestAssuredConfig;
    import io.restassured.specification.RequestSpecification;
    import org.junit.jupiter.api.Test;
    import static io.restassured.RestAssured.given;

    public class UpdateCatalogTest {

    @Test
    public void myTest() {
        getRequestSpecification().get("https://api.kraken.com/0/public/Ticker?pair=xbtusd");
    }

    RequestSpecification getRequestSpecification() {

        Options options = Options.builder()
                .build();

        RestAssuredConfig config = CurlRestAssuredConfigFactory.createConfig(options);

        RequestSpecification rq = given()
                .config(config)
                .when();

        return rq;
     }

    }

However, if I append the httpClient config not to reuse instances there's no more printing.

So having this:

        RestAssuredConfig config = CurlRestAssuredConfigFactory.createConfig(options).httpClient(HttpClientConfig.httpClientConfig().dontReuseHttpClientInstance());

instead of:

         RestAssuredConfig config = CurlRestAssuredConfigFactory.createConfig(options);

These are my pom versions:

     <parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>2.1.3.RELEASE</version>
    </parent>

    <properties>
        <rest-assured.version>4.2.0</rest-assured.version>
        <junit-jupiter.version>5.5.2</junit-jupiter.version>
        <gson.version>2.8.6</gson.version>
        <http.client.version>4.5.3</http.client.version>
        <curl.logger.version>2.0.0</curl.logger.version>
    </properties>

And this is my logback file:
```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
        <encoder>
            <pattern>%-4relative [%thread] %-5level %logger{35} - %msg %n</pattern>
        </encoder>
    </appender>
    <appender name="FILE" class="ch.qos.logback.core.FileAppender">
        <file>my-rest-assured-curl-log.log</file>
        <append>true</append>
        <encoder>
            <pattern>%d{HH:mm:ss.SSS} [%-5level] %logger{15} - %msg%n%rEx</pattern>
        </encoder>
    </appender>
    <logger name="curl" level="DEBUG">
        <appender-ref ref="STDOUT"/>
    </logger>
</configuration>
```

Thank you.

**Comment by dzieciou** [comment]
Unfortunately, that's the way it works.

curl-logger uses a custom HTTPClient factory that can create HTTP clients able to log curl commands. 

if you call `.dontReuseHttpClientInstance())` it resets REST-assured configuration to use REST-assured HTTP client factory that does not have this ability.

What is your ultimate goal? Why are you trying to modify the config?

**Comment by francislainy** [comment]
> Unfortunately, that's the way it works.
> 
> curl-logger uses a custom HTTPClient factory that can create HTTP clients able to log curl commands.
> 
> if you call `.dontReuseHttpClientInstance())` on REST-assured configuration created/updated by curl-logger it resets that configuration to use default REST-assured HTTP client factory that does not have this ability.
> 
> What is your ultimate goal? Why are you trying to modify the config?

Hi, sorry for the delay and thank you for your reply. The reason I had this there is because I see the error Invalid use of BasicClientConnManager: connection still allocated error that still comes up every now and then within my tests.

**Comment by dzieciou** [comment]
That has been fixed in v2.0.0 (by #37 ). In fact, the following test passes:

```java
package com.github.dzieciou.testing.curl;

import io.restassured.config.RestAssuredConfig;
import org.testng.annotations.Test;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.equalTo;

public class Try {

    @Test
    public void test() {
        RestAssuredConfig config = CurlRestAssuredConfigFactory.createConfig();
        assertThat(config.getHttpClientConfig().isConfiguredToReuseTheSameHttpClientInstance(),
                equalTo(false));
    }

}
```

**Comment by francislainy** [comment]
That's actua

> That has been fixed in v2.0.0 (by #37 ) and by default `createConfig()` creates configurations that do not reuse instances of HTTP client. In fact, the following test passes:
> 
> ```java
> package com.github.dzieciou.testing.curl;
> 
> import io.restassured.config.RestAssuredConfig;
> import org.testng.annotations.Test;
> 
> import static org.hamcrest.MatcherAssert.assertThat;
> import static org.hamcrest.Matchers.equalTo;
> 
> public class Try {
> 
>     @Test
>     public void test() {
>         RestAssuredConfig config = CurlRestAssuredConfigFactory.createConfig();
>         assertThat(config.getHttpClientConfig().isConfiguredToReuseTheSameHttpClientInstance(),
>                 equalTo(false));
>     }
> 
> }
> ```

We're actually already on the 2.0.0 version (4.2.0 for RestAssured). The issue seems to happen only from Jenkins within our nightly builds and not in a repeatable way but often enough so we're having a conditional statement to check if jenkins then we add the `dontReuseHttpClientInstance()` piece, which blocks the curl from printing.

**Comment by dzieciou** [comment]
Ok. I was able to reproduce your issue with cul-logger:2.0.0

>  Invalid use of BasicClientConnManager: connection still allocated 

This has been fixed 2.0.1 (not 2.0.0, as I said previously, my bad). Once I upgraded to 2.0.1 the problem disappeared. Try to upgrade to 2.0.1 at least.
